src/app/shared/components/filters/filters.component.ts

```diff
@@ -50,18 +60,36 @@ export class FiltersComponent {
     return FilterType;
   }
 
-  public get isFiltered() {
+  public get areFiltersChangedSinceLastApply(): boolean {
+    return this._areFiltersChangedSinceLastApply;
+  }
+
+  public get isFilterNotEmpty() {
     return !!this._filters.find(f => f.value);
   }
 
+  public ngOnChanges(changes: SimpleChanges): void {
+    if (changes['filters']) {
```

Comment:
#nit: `'someProp' in changes` is more truly than your statement, because `changes['someProp']` looks at value, but not on existings of it.